twisted/twisted
# [trac:5310] Make t.web.client.RequestGenerationFailed a public api.

Scope
=====

See: https://twistedmatrix.com/trac/ticket/5310


Changes
=======

Changes are based on https://github.com/twisted/twisted/compare/trunk...agent-exceptions-5310

I went for a different formatting for long imports, as I find that sorting them in alphabetic order and having a single import per line result in cleaner diffs

How to test
========

here is an example of using the new public export 
```
from twisted.internet import reactor
from twisted.web.client import Agent, RequestGenerationFailed

agent = Agent(reactor)
class Whatever(object):
    """
    I should probably implement some interfaces or something.
    """

def eb_handle(failure):
    """
    The actual traceback is found in the failure list.
    """
    failure.trap(RequestGenerationFailed)
    failure.value.reasons[0].printTraceback()

def bb_done(it):
    reactor.stop()

d = agent.request("POST", "http://example.com/", bodyProducer=Whatever())
d.addErrback(eb_handle)
d.addBoth(bb_done)

reactor.run()
```
